009-quarkus-infinispan-grpc-kafka/src/test/java/io/quarkus/qe/grateful/shutdown/KafkaGratefulShutdownTest.java

```diff
@@ -0,0 +1,122 @@
+package io.quarkus.qe.grateful.shutdown;
+
+import static io.restassured.RestAssured.post;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.logmanager.Level;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.LogFile;
+import io.quarkus.test.QuarkusProdModeTest;
+
+public class KafkaGratefulShutdownTest {
+
+    private static final int TOTAL_MESSAGES = 10;
+    private static final String GRATEFUL_SHUTDOWN_PROPERTY = "mp.messaging.incoming.slow.graceful-shutdown";
+    private static final String KAFKA_LOG_PROPERTY = "quarkus.log.category.\"io.smallrye.reactive.messaging.kafka\".level";
+    private static final String LAST_MESSAGE_LOG = "Processed Message " + TOTAL_MESSAGES;
+    private static final String SHUTDOWN_KAFKA_LOG = "Shutting down - Waiting for message processing to complete";
+
+    @RegisterExtension
+    static QuarkusProdModeTest app = new QuarkusProdModeTest()
+            .setArchiveProducer(
+                    () -> ShrinkWrap.create(JavaArchive.class)
+                            .addClasses(SlowTopicResource.class, SlowTopicConsumer.class))
+            .setRuntimeProperties(Collections.singletonMap("quarkus.http.port", "8083"))
```

Comment:
I don't know why, but using the default port 8081 is causing port conflict. I had to use another.

Reply:
I don't know either, Is something that we should have a look at. 
If you run this test: `mvn -Pnative -Dtest=GreetingServiceTest#testBlockingGreetingResource test`
And monitor 8081 port: `watch -n1 "sudo netstat -tulpn | grep LISTEN | grep 8081"`

You will see that is used by `@QuarkusTest` (that will launch the app in the default `quarkus.http.test-port`)

But If you run this PR test: `mvn -Dtest=KafkaGratefulShutdownTest#shouldWaitForMessagesWhenGratefulShutdownIsEnabled test`

Then the port `8081` is not used. Indeed if you just run this test you can swap to `8081`

Reply:
Georgios asked to stop using this extension the way we're using it, that's way I'm reluctant to investigate more these issues: we should stop using this tool :/